File: backend/src/main/kotlin/com/example/backend/models/Produit1.kt
package com.example.backend.models

import jakarta.persistence.*
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "produit1")
class Produit1 {
@Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "ean13", nullable = false, length = 16)
  var ean13: String? = null

  @Column(name = "reference", nullable = false, length = 32)
  var reference: String? = null

  @Column(name = "nom", length = 32)
  var nom: String? = null

  @Column(name = "contenance", nullable = false)
  var contenance: String? = null

  @Column(name = "stock")
  var stock: Int? = null

  @Column(name = "stock_max")
  var stockMax: Int? = null

  @Column(name = "stock_min")
  var stockMin: Int? = null

  @Column(name = "date_peremption")
  var datePeremption: LocalDateTime? = null

  @Column(name = "date_cmd")
  var dateCmd: LocalDateTime? = null

  @Column(name = "stock_mag")
  var stockMag: Int? = null

  @Column(name = "prix_public")
  var prixPublic: Double? = null

  @Column(name = "prix_achat")
  var prixAchat: Double? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "categorie_id")
  var categorie: Categorie? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "forme_id")
  var forme: Forme? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "fabriquant_id")
  var fabriquant: Fabriquant? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "fournisseur_id")
  var fournisseur: Fournisseur? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "rayon_id")
  var rayon: com.example.backend.models.Rayon? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "magasin_id")
  var magasin: Magasin? = null

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "unite_id")
  var unite: com.example.backend.models.Unite? = null
}

File: backend_py/pharmaPython/app/models/produit1.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.core.db import Base

class Produit1(Base):
  __tablename__ = "produit1"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True)
  ean13 = Column(String(16), nullable=False)
  reference = Column(String(32), nullable=False)
  nom = Column(String(32))
  contenance = Column(String, nullable=False)
  stock = Column(Integer)
  stock_max = Column(Integer)
  stock_min = Column(Integer)
  date_peremption = Column(DateTime)
  date_cmd = Column(DateTime)
  stock_mag = Column(Integer)
  prix_public = Column(Float, default=0.0)
  prix_achat = Column(Float, default=0.0)
  categorie_id = Column(Integer, ForeignKey("categorie.id"))
  forme_id = Column(Integer, ForeignKey("forme.id"))
  fabriquant_id = Column(Integer, ForeignKey("fabriquant.id"))
  fournisseur_id = Column(Integer, ForeignKey("fournisseur.id"))
  rayon_id = Column(Integer, ForeignKey("rayon.id"))
  magasin_id = Column(Integer, ForeignKey("magasin.id"))
  unite_id = Column(Integer, ForeignKey("unite.id"))

  categorie = relationship("Categorie")
  forme = relationship("Forme")
  fabriquant = relationship("Fabriquant")
  fournisseur = relationship("Fournisseur")
  rayon = relationship("Rayon")
  magasin = relationship("Magasin")
  unite = relationship("Unite")
